x-pack/plugins/security_solution/public/flyout/right/hooks/use_fetch_unique_hosts.ts

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useQuery } from '@tanstack/react-query';
+import type { IEsSearchRequest } from '@kbn/data-plugin/common';
+import { createFetchAggregatedData } from '../utils/fetch_aggregated_data';
+import { useKibana } from '../../../common/lib/kibana';
+
+const AGG_KEY = 'uniqueHosts';
+const QUERY_KEY = 'useFetchUniqueHosts';
+
+interface RawAggregatedDataResponse {
+  aggregations: {
+    [AGG_KEY]: {
+      buckets: unknown[];
+    };
+  };
+}
+
+const searchRequest: IEsSearchRequest = {
+  params: {
+    body: {
+      aggs: {
+        [AGG_KEY]: {
+          terms: {
+            field: 'host.name',
+            size: 1000,
```

Comment:
nit: consider making this a constant. 

I think the query size of 1000 started as a one off in the [tree map visualizations](https://github.com/elastic/kibana/pull/126896/files#diff-32be31be9b63ef368f0dff020a1839cbc801ac92390f83e5ba9b9e0d6e5630c4R11), and then used in KPI viz and now flyout. maybe worth making it a constant in `x-pack/plugins/security_solution/common/constants.ts`?

Reply:
I will integrate this change on my coming PR, as it refactors this code and moves it to the shared folder anyway!